### PR TITLE
Fix move constructor docstring

### PR DESCRIPTION
**Pull request by alaeddine-13**
move docstrings to be under constructor definition and keep an overview under each executor class definition

### CODE REVIEW

**Comment by nan-wang** [comment]
You might want to update with the latest `main` branch before moving on

File: jinahub/encoders/image/BigTransferEncoder/big_transfer.py [hunk 1]
```diff
@@ -67,6 +39,34 @@ def __init__(
...
+        """
+        :param model_path: the path of the model in the `SavedModel` format.
+        :param model_name: includes ``Imagenet1k/R50x1``, ``Imagenet1k/R101x1``,
+            ``Imagenet1k/R50x3``, ``Imagenet1k/R101x3``, ``Imagenet1k/R152x4``,
+            ``Imagenet21k/R50x1``, ``Imagenet21k/R101x1``, ``Imagenet21k/R50x3``,
+            ``Imagenet21k/R101x3``, ``Imagenet21k/R152x4``
+
+        This encoder checks if the specified model_path exists.
+        If it does exist, the model in this folder is used.
+        If it does not exist, the model specified in the model_name will be
```
Comment (nan-wang) [inline jinahub/encoders/image/BigTransferEncoder/big_transfer.py:51]:
```suggestion
        If it does not exist, the model specified in the `model_name` will be
```

File: jinahub/encoders/image/BigTransferEncoder/big_transfer.py [hunk 2]
```diff
@@ -67,6 +39,34 @@ def __init__(
...
     ):
+        """
+        :param model_path: the path of the model in the `SavedModel` format.
+        :param model_name: includes ``Imagenet1k/R50x1``, ``Imagenet1k/R101x1``,
+            ``Imagenet1k/R50x3``, ``Imagenet1k/R101x3``, ``Imagenet1k/R152x4``,
+            ``Imagenet21k/R50x1``, ``Imagenet21k/R101x1``, ``Imagenet21k/R50x3``,
+            ``Imagenet21k/R101x3``, ``Imagenet21k/R152x4``
+
+        This encoder checks if the specified model_path exists.
+        If it does exist, the model in this folder is used.
```
Comment (nan-wang) [inline jinahub/encoders/image/BigTransferEncoder/big_transfer.py:50]:
```suggestion
        If it exists, the model in this folder is used.
```

File: jinahub/encoders/image/BigTransferEncoder/big_transfer.py [hunk 3]
```diff
@@ -67,6 +39,34 @@ def __init__(
...
+        which has the following structure:
+
+        .. highlight:: bash
+         .. code-block:: bash
+            .
+            ├── saved_model.pb
+            └── variables
+                ├── variables.data-00000-of-00001
+                └── variables.index
+        :param: device: Device ('/CPU:0', '/GPU:0', '/GPU:X')
```
Comment (nan-wang) [inline jinahub/encoders/image/BigTransferEncoder/big_transfer.py:64]:
```suggestion
        :param device: Device ('/CPU:0', '/GPU:0', '/GPU:X')
```

**Review by nan-wang — CHANGES_REQUESTED** [review]

**Review by nan-wang — DISMISSED** [review]
LGTM👍

File: jinahub/encoders/image/CustomImageTorchEncoder/custom_image_torch_encoder.py [hunk 4]
```diff
@@ -52,6 +38,22 @@ def __init__(
...
+        :param layer_name: The layer name from which to extract the feature maps.
+            These feature maps will then be fed into an `AdaptiveAvgPool2d` layer
+        to extract the embeddings
+        :param device: The device where to load the model.
+        :param default_batch_size: fallback batch size in case there is not batch size
+            sent in the request
+        :param default_traversal_paths: fallback traversal path in case there is no
+            traversal path sent in the request
+        :param args:  Additional positional arguments.
+        :param kwargs: Additional keyword arguments.
```
[the remark this answers is not in the document]
Comment (alaeddine-13) [reply jinahub/encoders/image/CustomImageTorchEncoder/custom_image_torch_encoder.py:55]: no they are not displayed so not needed. 
To be consistent, I'll remove all the args and kwargs from executors docstrings

**Comment by alaeddine-13** [comment]
args and kwargs removed
can you re-check ?

File: jinahub/crafters/ImageNormalizer/normalizer.py [hunk 5]
```diff
@@ -24,7 +26,19 @@ def __init__(
         *args,
         **kwargs,
     ):
-        """Set Constructor."""
+        """
+        `ImageNormalizer` resizes, crops and normalizes images stored in Document blobs.
```
Comment (cristianmtr) [inline jinahub/crafters/ImageNormalizer/normalizer.py:30]: Is it necessary to have the class description in the init docstring?

File: jinahub/encoders/image/CLIPImageEncoder/clip_image.py [hunk 6]
```diff
@@ -1,123 +1,122 @@
...
+from typing import Optional, Tuple
+
+import torch
+from jina import DocumentArray, Executor, requests
+from jina.logging.logger import JinaLogger
+from jina_commons.batching import get_docs_batch_generator
+from transformers import CLIPFeatureExtractor, CLIPModel
+
+
+class CLIPImageEncoder(Executor):
```
Comment (cristianmtr) [inline jinahub/encoders/image/CLIPImageEncoder/clip_image.py:10]: The diff for this file looks odd.
Screenshot from 2021-09-16 15-41-53

File: jinahub/encoders/image/ImageTFEncoder/image_tf_encoder.py [hunk 7]
```diff
@@ -89,7 +88,7 @@ def encode(self, docs: DocumentArray, parameters: Dict, **kwargs):
         B` is the batch size and `D` is the Dimension.
 
         :param docs: DocumentArray containing blob as image data.
-        :param args: additional positional arguments.
+        :param parameters: parameters dictionary.
         :param kwargs: additional positional arguments.
```
Comment (cristianmtr) [inline jinahub/encoders/image/ImageTFEncoder/image_tf_encoder.py:92]:
```suggestion
        :param kwargs: additional keyword arguments.
```

File: jinahub/encoders/text/AudioCLIPTextEncoder/executor/audioclip_text.py [hunk 8]
```diff
@@ -33,6 +25,15 @@ def __init__(
         *args,
         **kwargs
     ):
+        """
+        :param model_path: path to the pre-trained AudioCLIP model.
+        :param default_traversal_paths: default traversal path (used if not specified in
+            request's parameters)
+        :param default_batch_size: default batch size (used if not specified in
+            request's parameters)
+        :param device: device that the model is on (should be "cpu", "cuda" or "cuda:X",
```
Comment (cristianmtr) [inline jinahub/encoders/text/AudioCLIPTextEncoder/executor/audioclip_text.py:34]: Also, why don't we document *args and **kwargs? Not a big deal, just wanted to check it's intentional and agreed upon.

File: jinahub/encoders/text/CLIPTextEncoder/clip_text.py [hunk 9]
```diff
@@ -1,90 +1,89 @@
...
-            padding='longest',
-            truncation=True,
-            return_tensors='pt',
-        )
-        input_tokens = {k: v.to(self.device) for k, v in input_tokens.items()}
-        return input_tokens
+from typing import Dict, Optional, Sequence
+
+import torch
+from jina import DocumentArray, Executor, requests
```
Comment (cristianmtr) [inline jinahub/encoders/text/CLIPTextEncoder/clip_text.py:4]: The diff looks odd here too. Like the entire file was deleted and rewritten.

File: jinahub/rankers/LightGBMRanker/lightgbm_ranker.py [hunk 10]
```diff
@@ -14,22 +14,9 @@ class LightGBMRanker(Executor):
...
-    :param params: Parameters used to train the LightGBM learning-to-rank model.
-    :param categorical_query_features: name of features contained in `query_features` corresponding to categorical features.
-    :param categorical_match_features: name of features contained in `match_features` corresponding to categorical features.
-    :param query_features_before: True if `query_features` must be placed before the `match` ones in the `dataset` used for prediction.
-    :param args: Additional positional arguments
-    :param kwargs: Additional keyword arguments
     .. note::
-        The name of the features are used to extract the features from incoming `documents`. Check how these features are accessed in
-        :class:`Document` at https://docs.jina.ai/api/jina.types.document/
+        The name of the features used by this Executor are used to extract the features from incoming `documents`.
```
Comment (cristianmtr) [inline jinahub/rankers/LightGBMRanker/lightgbm_ranker.py:18]: Why is this here? It should be under `features` docstring, no?

File: jinahub/segmenters/Sentencizer/sentencizer.py [hunk 11]
```diff
@@ -38,7 +27,22 @@ def __init__(
         *args,
         **kwargs
     ):
-        """Set constructor."""
+        """
+        :class:`Sentencizer` split the text on the doc-level
```
Comment (cristianmtr) [inline jinahub/segmenters/Sentencizer/sentencizer.py:31]: no docs on the class under init

File: jinahub/segmenters/TorchObjectDetectionSegmenter/torch_object_detection_segmenter.py [hunk 12]
```diff
@@ -138,7 +128,19 @@ def __init__(
         *args,
         **kwargs,
     ):
-        """Set constructor"""
+        """
+        :class:`TorchObjectDetectionSegmenter` detects objects
```
Comment (cristianmtr) [inline jinahub/segmenters/TorchObjectDetectionSegmenter/torch_object_detection_segmenter.py:132]: same

File: jinahub/segmenters/VADSpeechSegmenter/vad_speech_segmenter.py [hunk 13]
```diff
@@ -31,6 +26,12 @@ def __init__(
         *args,
         **kwargs,
     ):
+        """
+        Segment the speech audio using Silero's Voice Activity Detector (VAD)
```
Comment (cristianmtr) [inline jinahub/segmenters/VADSpeechSegmenter/vad_speech_segmenter.py:30]: same

**Review by cristianmtr — CHANGES_REQUESTED** [review]

File: jinahub/encoders/image/ImageTFEncoder/image_tf_encoder.py (hunk 7 above)
Comment (alaeddine-13) [reply jinahub/encoders/image/ImageTFEncoder/image_tf_encoder.py:92]: forgot to remove kwargs too

File: jinahub/encoders/text/AudioCLIPTextEncoder/executor/audioclip_text.py (hunk 8 above)
Comment (alaeddine-13) [reply jinahub/encoders/text/AudioCLIPTextEncoder/executor/audioclip_text.py:34]: we don't display them in the hub website
so for hub they are not needed but having them will not cause a problem
from the pep conventions, not having them in the docstrings is not considered a breach of pep rules and will not fail linting
Nan said that "adding duplicated codes will definitely increases our maintenance effort" so maybe that's an argument to not include them

File: jinahub/crafters/ImageNormalizer/normalizer.py (hunk 5 above)
Comment (alaeddine-13) [reply jinahub/crafters/ImageNormalizer/normalizer.py:30]: no it's not necessary
I made research on what is the best way to write the docstrings
It's not clear if an overview of what a constructor does can be included under the constructor.
Currently, we dont' display the docstring, it's just used to extract the description of the parameters

Comment (alaeddine-13) [reply jinahub/crafters/ImageNormalizer/normalizer.py:30]: So for convenience, I'll keep only parameters in the docstring

File: jinahub/rankers/LightGBMRanker/lightgbm_ranker.py (hunk 10 above)
Comment (alaeddine-13) [reply jinahub/rankers/LightGBMRanker/lightgbm_ranker.py:18]: This applies for all the *_features parameters in the constructor.
If I put it in the constructor docstring, it will not be displayed.
I can't put it after the parameters because they will show up in one of the parameter's description.
I'll improve the docstringso that it's clear we're referring to the constructor parameters

File: jinahub/encoders/image/CLIPImageEncoder/clip_image.py (hunk 6 above)
Comment (alaeddine-13) [reply jinahub/encoders/image/CLIPImageEncoder/clip_image.py:10]: according to pycharm, the difference is the new line character (one is CRLF and the other is LF)
image
This appeared when I merged main to the branch:
https://github.com/jina-ai/executors/pull/203/commits/bc4422d29faaea4354193192d3996b113875cfdb

Comment (alaeddine-13) [reply jinahub/encoders/image/CLIPImageEncoder/clip_image.py:10]: I suggest we keep it as it is since LF is the new line character in linux
maybe this happened when someone worked on a different OS

File: jinahub/encoders/text/CLIPTextEncoder/clip_text.py (hunk 9 above)
Comment (alaeddine-13) [reply jinahub/encoders/text/CLIPTextEncoder/clip_text.py:4]: explained above

File: jinahub/crafters/ImageNormalizer/normalizer.py (hunk 5 above)
Comment (alaeddine-13) [reply jinahub/crafters/ImageNormalizer/normalizer.py]: done

**Review by cristianmtr — APPROVED** [review]
